docs/resources/server/datamaps/index.md

```diff
@@ -0,0 +1,390 @@
+# Data Maps
+
+A data map contains data-driven, reloadable objects that can be attached to a registered object. This system allows for more easily data-driving game behaviour, as they provide functionality such as syncing or conflict resolution, leading to a better and more configurable user experience. You can think of [tags] as registry object ➜ boolean maps, while data maps are more flexible registry object ➜ object maps. Similar to [tags], data maps will add to their corresponding data map rather than overwriting.
+
+Data maps can be attached to both static, built-in, registries and dynamic data-driven datapack registries. Data maps support reloading through the use of the `/reload` command or any other means that reload server resources.
+
+NeoForge provides various [built-in data maps][builtin] for common use cases, replacing hardcoded vanilla fields. More info can be found in the linked article.
+
+## File Location
+
+Data maps are loaded from a JSON file located at `<mapNamespace>/data_maps/<registryNamespace>/<registryPath>/<mapPath>.json`, where:
+
+- `<mapNamespace>` is the namespace of the ID of the data map,
+- `<mapPath>` is the path of the ID of the data map,
+- `<registryNamespace>` is the namespace of the ID of the registry (omitted if it is `minecraft`), and
+- `<registryPath>` is the path of the ID of the registry.
+
+Examples:
+
+- For a data map named `mymod:drop_healing` for the `minecraft:item` registry (as in the example below), the path will be `mymod/data_maps/item/drop_healing.json`.
+- For a data map named `somemod:somemap` for the `minecraft:block` registry, the path will be `somemod/data_maps/block/somemap.json`.
+- For a data map named `example:stuff` for the `somemod:custom` registry, the path will be `example/data_maps/somemod/custom/stuff.json`.
+
+## JSON Structure
+
+A data map file itself may contain the following fields:
+
+- `replace`: A boolean that will clear the data map before adding the values of this file. This should never be shipped by mods, and only be used by pack developers that want to overwrite this map for their own purposes.
+- `neoforge:conditions`: A list of [loading conditions][conditions].
+- `values`: A map of registry IDs or tag IDs to values that should be added to the data map by your mod. The structure of the values themselves is defined by the data map's codec (see below).
+- `remove`: A list of registry IDs or tag IDs to be removed from the data map.
+
+### Adding Values
+
+For example, let's assume that we have a data map object with two float keys `amount` and `chance` for the registry `minecraft:item`. A corresponding data map file could look something like this:
+
+```json5
```

Comment:
Only going to mention this once, but all code blocks should be 4 spaces for tabs.